lib/services/access_monitoring_rules.go

```diff
@@ -39,6 +42,45 @@ type AccessMonitoringRules interface {
 	ListAccessMonitoringRules(ctx context.Context, limit int, startKey string) ([]*accessmonitoringrulesv1.AccessMonitoringRule, string, error)
 }
 
+// NewAccessMonitoringRuleWithLabels creates a new AccessMonitoringRule  with the given spec and labels.
+func NewAccessMonitoringRuleWithLabels(name string, labels map[string]string, spec *accessmonitoringrulesv1.AccessMonitoringRuleSpec) (*accessmonitoringrulesv1.AccessMonitoringRule, error) {
+	amr := &accessmonitoringrulesv1.AccessMonitoringRule{
+		Kind:    types.KindAccessMonitoringRule,
+		Version: types.V1,
+		Metadata: &headerv1.Metadata{
+			Name:      name,
+			Namespace: defaults.Namespace,
+			Labels:    labels,
+		},
+		Spec: spec,
+	}
+
+	err := ValidateAccessMonitoringRule(amr)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return amr, nil
+}
+
+// ValidateAccessMonitoringRule checks that the provided access monitoring rule is valid.
+func ValidateAccessMonitoringRule(accessMonitoringRule *accessmonitoringrulesv1.AccessMonitoringRule) error {
```

Comment:
what about the following:

- shouldn't `spec.condition` always be defined? 
- if `spec.notification` defined, is at least one `recipient` defined and `name` is defined?
  - if a `subjects` include `access_request`, does `spec.notification` required to be defined?

Reply:
There are use cases where the recipients make sense to be empty since not all plugins use recipients. I don't think we should enforce notification being required if the subjects contains access_request since someone might want to  use the state changing part instead of the notification functionality.
I added in the checks for condition and notification.name

Reply:
>  I don't think we should enforce notification being required if the subjects contains access_request since someone might want to use the state changing part instead of the notification functionality.

i double checked with @r0mant, and he agrees that we should tighten up the validation for `subjects including access_requeset` (this is what we do in the UI). he said we can relax it later if it becomes an issue for some reason

Reply:
also, do you remember which plugins don't accept recipients? i might have to account this in the UI

Reply:
I know servicenow doesn't use recipients, pretty sure the rest do use them in different ways.

Reply:
Have added in the check for notification if the any of the subjects is an access request